contracts/src/v0.8/functions/dev/1_0_0/Functions.sol

```diff
@@ -0,0 +1,147 @@
+
+    CBOR.writeString(buffer, "language");
+    CBOR.writeUInt256(buffer, uint256(self.language));
+
+    CBOR.writeString(buffer, "source");
+    CBOR.writeString(buffer, self.source);
+
+    if (self.args.length > 0) {
+      CBOR.writeString(buffer, "args");
+      CBOR.startArray(buffer);
+      for (uint256 i = 0; i < self.args.length; i++) {
+        CBOR.writeString(buffer, self.args[i]);
+      }
+      CBOR.endSequence(buffer);
+    }
+
+    if (self.secrets.length > 0) {
+      if (self.secretsLocation == Location.Inline) {
+        revert NoInlineSecrets();
+      }
+      CBOR.writeString(buffer, "secretsLocation");
+      CBOR.writeUInt256(buffer, uint256(self.secretsLocation));
+      CBOR.writeString(buffer, "secrets");
+      CBOR.writeBytes(buffer, self.secrets);
+    }
+
+    return buffer.buf.buf;
+  }
+
+  /**
+   * @notice Initializes a Chainlink Functions Request
+   * @dev Sets the codeLocation and code on the request
+   * @param self The uninitialized request
+   * @param location The user provided source code location
+   * @param language The programming language of the user code
+   * @param source The user provided source code or a url
+   */
+  function initializeRequest(
+    Request memory self,
+    Location location,
+    CodeLanguage language,
+    string memory source
+  ) internal pure {
+    if (bytes(source).length == 0) revert EmptySource();
+
+    self.codeLocation = location;
+    self.language = language;
+    self.source = source;
+  }
+
+  /**
+   * @notice Initializes a Chainlink Functions Request
+   * @dev Simplified version of initializeRequest for PoC
+   * @param self The uninitialized request
+   * @param javaScriptSource The user provided JS code (must not be empty)
+   */
+  function initializeRequestForInlineJavaScript(Request memory self, string memory javaScriptSource) internal pure {
+    initializeRequest(self, Location.Inline, CodeLanguage.JavaScript, javaScriptSource);
+  }
+
+  /**
+   * @notice Adds Remote user encrypted secrets to a Request
+   * @param self The initialized request
+   * @param encryptedSecretsURLs Encrypted comma-separated string of URLs pointing to off-chain secrets
+   */
+  function addRemoteSecrets(Request memory self, bytes memory encryptedSecretsURLs) internal pure {
+    if (encryptedSecretsURLs.length == 0) revert EmptySecrets();
+
+    self.secretsLocation = Location.Remote;
+    self.secrets = encryptedSecretsURLs;
+  }
+
+  /**
+   * @notice Adds args for the user run function
+   * @param self The initialized request
+   * @param args The array of args (must not be empty)
+   */
+  function addArgs(Request memory self, string[] memory args) internal pure {
+    if (args.length == 0) revert EmptyArgs();
+
+    self.args = args;
+  }
+
+  /**
+   * @notice Add request data version to the request CBOR 
+   */
+  function encodeRequest(bytes memory requestCBOR) internal pure returns (bytes memory) {
```

Comment:
What's the benefit of having it in another layer around the message data vs. inside a CBOR field?

Reply:
None really, cleaned this up in favor of just providing it as an arg to `sendRequest`

Reply:
👍